### DISCUSSION TITLE
Color Correction

### DISCUSSION THREAD

**Discussion by simon-cross** [openMVG/openMVG#2020] [Q&A] [answered]
I have read in the paper "Efficient global color correction for large-scale multiple-view images in three-dimensional reconstruction", that openMVG implements a color matching approach for multi-view images. How is it possible to run openMVG and get the color corrected images?

**Comment by pmoulon** [comment, marked as answer]
Sorry for my belated answer.
You can find the description on how to run the tools here https://github.com/openMVG/Color_Harmonization_Image_Dataset
Perhaps a bit outdated but should be easy to adapt to new command line tools.

**Reply by simon-cross** [reply]
Thank you for your reply.

I have tried the commands listed on the webpage of the link in your reply, but I get stuck at the third command.

```
openMVG_main_ComputeMatches -i /home/simone/Repository/ReconstructionOptimizationFinal/Data/Volo001/2020_05_Noho_session_2_Rafa_shortdance_take1/Frame_00001/ColorCorrection/Frames_Matches/sfm_data.json -o /home/simone/Repository/ReconstructionOptimizationFinal/Data/Volo001/2020_05_Noho_session_2_Rafa_shortdance_take1/Frame_00001/ColorCorrection/Frames_Matches/matches.txt  -p /home/simone/Repository/ReconstructionOptimizationFinal/Data/Volo001/2020_05_Noho_session_2_Rafa_shortdance_take1/Frame_00001/ColorCorrection/Frames_Matches/pairs.txt
INFO: [main_ComputeMatches.cpp:114]  You called : 
openMVG_main_ComputeMatches
--input_file /home/simone/Repository/ReconstructionOptimizationFinal/Data/Volo001/2020_05_Noho_session_2_Rafa_shortdance_take1/Frame_00001/ColorCorrection/Frames_Matches/sfm_data.json
--output_file /home/simone/Repository/ReconstructionOptimizationFinal/Data/Volo001/2020_05_Noho_session_2_Rafa_shortdance_take1/Frame_00001/ColorCorrection/Frames_Matches/matches.txt
--pair_list /home/simone/Repository/ReconstructionOptimizationFinal/Data/Volo001/2020_05_Noho_session_2_Rafa_shortdance_take1/Frame_00001/ColorCorrection/Frames_Matches/pairs.txt
Optional parameters:
--force 0
--ratio 0.8
--nearest_matching_method AUTO
--cache_size unlimited
--preemptive_feature_used/count 0 / 200
INFO: [loggerprogress.hpp:79] [- Regions Loading -] 50%
INFO: [loggerprogress.hpp:79] [- Regions Loading -] 100%
INFO: [main_ComputeMatches.cpp:217]  - PUTATIVE MATCHES - 
INFO: [main_ComputeMatches.cpp:238] Using FAST_CASCADE_HASHING_L2 matcher
ERROR: [Pair_Builder.hpp:56] loadPairs: Impossible to read the specified file: "/home/simone/Repository/ReconstructionOptimizationFinal/Data/Volo001/2020_05_Noho_session_2_Rafa_shortdance_take1/Frame_00001/ColorCorrection/Frames_Matches/pairs.txt".
ERROR: [main_ComputeMatches.cpp:314] Failed to load pairs from file: "/home/simone/Repository/ReconstructionOptimizationFinal/Data/Volo001/2020_05_Noho_session_2_Rafa_shortdance_take1/Frame_00001/ColorCorrection/Frames_Matches/pairs.txt"
```

The error message says that it is not possible to read the file pairs.txt. But this file, is an output file. Why should the code need to read an output file?

**Reply by pmoulon** [reply]
Could you try and ignore the pairs.txt file from the command line

**Reply by simon-cross** [reply]
Without pairs.txt the command runs without errors. But, now I get an error message when running `openMVG_main_ColHarmonize`:
```
Global Color Harmonization

sh: 1: neato: not found
sh: 1: neato: not found
sh: 1: neato: not found
INFO: [colorHarmonizeEngineGlobal.cpp:529] 
ColorHarmonizationEngineGlobal::CleanGraph() :: => connected Component cardinal: 1
sh: 1: neato: not found
INFO: [colorHarmonizeEngineGlobal.cpp:589] 
Cardinal of nodes: 2
Cardinal of edges: 1
INFO: [colorHarmonizeEngineGlobal.cpp:195] 
 Remaining cameras after CC filter : 
2 from a total of 12
INFO: [colorHarmonizeEngineGlobal.cpp:222] Current edge : FVV_0_001_00001.png	FVV_0_003_00001.png
INFO: [colorHarmonizeEngineGlobal.cpp:329] 
 -- 
 SOLVE for color consistency with linear programming
 --
openMVG_main_ColHarmonize: /home/simone/Repository/ReconstructionOptimizationFinal/Code/ThirdParty/openMVG/src/openMVG/linearProgramming/linearProgrammingOSI_X.cpp:120: virtual bool openMVG::linearProgramming::OSI_X_SolverWrapper::setup(const openMVG::linearProgramming::LP_Constraints_Sparse&): Assertion `nbParams_ == cstraints.nbParams_' failed.
Aborted (core dumped)
```

**Reply by rperrot** [reply]
It seems that graphviz is not installed on your computer.

**Reply by pmoulon** [reply]
Seems like you have only 2 image, is it right?

If you install graphviz you would see some files showing the image graph you are trying to process.

If you don't mind feel free to share your images with us and we can try to solve what is happening.
Also did you try the image samples we are providing?

**Comment by simon-cross** [comment]
Hi,

Thank you for your reply. In the attachment, there are the images that I am
trying to process. There are two folders, one with the images having the
same resolution, and the other with images with different resolutions.

Regards,
Simone
 Frames.zip
<https://drive.google.com/file/d/1B6T-tXa_ze62M8jX1mp47NP9Kw0jWR3j/view?usp=drive_web>

**Comment by simon-cross** [comment]
I have just allowed you to download the frames I sent you.

**Comment by simon-cross** [comment]
Hi,

have you had the chance to look at my problem?

Regards,
Simone

**Comment by simon-cross** [comment]
Hi,

Have you had the chance to look at my problem?

Regards,
Simone

**Comment by pmoulon** [comment]
Sorry @simon-cross I was not able to look at it yet. I got some serious cold symptoms on me and my family members and I'm unable to look to this for some time

**Comment by simon-cross** [comment]
No problem Pierre. Get well soon!

**Reply by pmoulon** [reply]
Thx you, I hope I will feel better soon too ;-)

**Comment by simon-cross** [comment]
Hi Pierre,

I hope you are well now.

I write to you because I haven't solved my problem yet. Have you had the
time to work on it?

Best,
Simone
